### ISSUE TITLE
mod_wsgi with WampServer and Anaconda 3 says module not found

### ISSUE THREAD

**Issue by dglund** [GrahamDumpleton/mod_wsgi#391]
Python: Anaconda 3 (Python 3.7.1)
WampServer 3.1.4 64 bit (includes apache 2.4.35)

I run into problems trying to launch Apache with mod_wsgi.

The first steps:
In CMD: `set "MOD_WSGI_APACHE_ROOTDIR=C:/wamp64/bin/apache/apache2.4.35”` 
In CMD: `pip install mod_wsgi` (successfully installs)
In CMD: `mod_wsgi-express module-config` 
which returns:
```
LoadFile "c:/users/[username]/anaconda3/python37.dll"
LoadModule wsgi_module "c:/users/[username]/anaconda3/lib/site-packages/mod_wsgi/server/mod_wsgi.cp37-win_amd64.pyd"
WSGIPythonHome "c:/users/[username]/anaconda3"
```
* WampServer runs fine in it's original form.
* I then add the following into the httpd.conf:
`LoadModule wsgi_module "c:/users/[username]/anaconda3/lib/site-packages/mod_wsgi/server/mod_wsgi.cp37-win_amd64.pyd"`
* Now Apache won't start with WampServer (2 of 3 services running)
* I open the Wamp menu and click through: Apache -> Apache modules, which shows a list of modules. At the very top, it shows wsgi_module as "no module file". When I click it, the PHP prompt opens and shows the exact following:
```
Sorry,
There is 'LoadModule wsgi_module modules/mod_wsgi.so' line in httpd.conf file
but there is no 'mod_wsgi.so' file in the apachex.y.z/modules/ directory.
Press ENTER to continue...
```

This seems odd to me, as I have never told apache to look for "modules/mod_wsgi.so" in the httpd.conf (i searched the file just in case). I have checked and confirmed that I added nothing more and nothing less than `LoadModule wsgi_module "c:/users/[username]/anaconda3/lib/site-packages/mod_wsgi/server/mod_wsgi.cp37-win_amd64.pyd"`
If I remove this line from httpd.conf, everything runs fine again. What could be causing this?

I have also tried adding 
`WSGIPythonHome "c:/users/[username]/anaconda3"` 
and 
`LoadFile "c:/users/[username]/anaconda3/python36.dll"`
... in various combinations without any difference.

**Comment by GrahamDumpleton** [comment]
You are meant to include all the lines output from ``module-config``, not just ``LoadModule``.

The user that the Apache service runs as, must have permission to access your user directories where things are installed.

**Comment by dglund** [comment]
I must have been missing one of the lines from the output, thank you. However, localhost/app only displays the folder structure of the web page, not the actual page.

**Comment by GrahamDumpleton** [comment]
As looks you might have worked out, you still need to setup ``WSGIScriptAlias`` to mount your WSGI application.

**Comment by dglund** [comment]
Reopening this as I noticed I was wrong, the mod_wsgi still cannot be found. But all wamp-services are running. The same error message for mod_wsgi persists though. I also added the WSGIScriptAlias line.
httpd.conf:

```
LoadFile "c:/users/[username]/anaconda3/python37.dll"
LoadModule wsgi_module "c:/users/[username]/anaconda3/lib/site-packages/mod_wsgi/server/mod_wsgi.cp37-win_amd64.pyd"
WSGIPythonHome "c:/users/[username]/anaconda3"
WSGIScriptAlias / "d:/project/developement/website/wsgi.py"
```
The same problem persists with or without the WSGIScriptAlias-line. 
Still can't figure out why Apache is looking for
`'mod_wsgi.so' file in the apachex.y.z/modules/ directory.`

**Comment by dglund** [comment]
Update:
Ok, so I tried uninstalling mod_wsgi (with pip uninstall) and Wamp, and installed Apache24 from ApacheLounge instead. Made sure everything is 64 bit.

With mod_wsgi uninstalled (confirmed that it was fully removed from /lib/site-packages/), I restarted the CMD and ran the set-command again, for the new Apache directory:

`set MOD_WSGI_APACHE_ROOTDIR="C:/Apache24”` (I tried various combinations of with/without quotation marks)

I then ran 
`pip install mod_wsgi` which successfully installed mod_wsgi.

However, when I (out of curiosity) open the apxs_config.py in mod_wsgi/server/ it says:
`LIBEXECDIR = 'C:\wamp64\bin\apache\apache2.4.35/lib'`
...Which makes me wonder. Since I uninstalled the mod, and ran a new set-command before reinstalling, how can this file remember the wamp-directory?

**Comment by dglund** [comment]
I realized that and got it fixed. I still face the problem with loading mod_wsgi though, even though the directory now seems to be fine. However I'm starting to suspect that the mod is actually being loaded despite of the module list in Apache saying it isn't. 

If I add the WSGIScriptAlias-line to the httpd.conf, I get an 500 Internal Server error in the browser, instead of the directory (which I suspect is due to misconfigurations in the Django project and wsgi.py) and the bottom line says:

`Apache/2.4.35 (Win64) PHP/7.2.10 mod_wsgi/4.6.5 Python/3.7 Server at myvirtualhost Port 80`

So I'll probably assume the mod is being loaded regardless of the Apache module list saying it isn't, and start working on the wsgi-configurations instead. Right now, I just moved the project from my local project directory to the virtual host directory, so there is probably some configurations to do.
